web/src/features/prompts/server/actions/createPrompt.ts

```diff
@@ -82,7 +83,16 @@ export const createPrompt = async ({
       })),
     );
 
+  // Lock and invalidate cache for _all_ versions and labels of the prompt name
+  const promptService = new PromptService(prisma, redis);
+  await promptService.lockCache({ projectId, promptName: name });
```

Comment:
Why did you decide to go with the locking strategy?
- if not locking, we would be serving wrong prompts from cache until ttl expires
- if locking, we could fail here and keep the lock forever